proxy/src/lib.rs

```diff
@@ -1,4 +1,109 @@
-#![deny(clippy::undocumented_unsafe_blocks)]
+// rustc lints/lint groups
+// https://doc.rust-lang.org/rustc/lints/groups.html
+#![deny(
+    deprecated,
+    future_incompatible,
+    // TODO: consider let_underscore
+    nonstandard_style,
+    rust_2018_idioms,
+    rust_2021_compatibility,
+    rust_2024_compatibility
```

Comment:
Are these three in any way mutually exclusive? They don't seem to overlap in what they enable nor do any of their lints conflict.

Reply:
Yeah, let's remove the first two.

Reply:
Done.